presto-raptor/src/main/java/com/facebook/presto/raptor/storage/organization/ShardCompactionManager.java

```diff
@@ -232,6 +232,10 @@ private boolean needsCompaction(ShardMetadata shard)
         if (shard.getRowCount() < (FILL_FACTOR * maxShardRows)) {
             return true;
         }
+
+        if (shard.getDeltaUuid().isPresent()) {
```

Comment:
`return shard.getDeltaUuid().isPresent();`

Reply:
I also thought about it when self review, but if you see the context of this code, if I change it this way, then anyone who want to add more condition to it, they need to change this code. So I think it's better to stay consistent